src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -1172,18 +1172,18 @@ void L2NormComputeEx(const nnvm::NodeAttrs& attrs,
 template<int ndim, bool clip = true>
 struct pick {
   template<typename DType, typename IType>
-  MSHADOW_XINLINE static void Map(int i, DType* out, const DType* a,
-                                  const IType *idx, int M, int stride,
+  MSHADOW_XINLINE static void Map(index_t i, DType* out, const DType* a,
+                                  const IType *idx, size_t M, int stride,
```

Comment:
Can we just use `index_t` for M so you don't need so many casting in this function?

Reply:
+1 for using index_t for M.

Reply:
Sure. i looked at other places where large array support was mentioned. It seemed that
`index_t` was used where the variable is used as index
and `size_t` was used otherwise. Hence I stuck to that. But I'll change it to avoid casting if that's what's useful.